docs/index.md

```diff
@@ -3,85 +3,326 @@ hide:
+  </div>
+</header>
+
+<section style="padding: 72px 0">
+  <div class="center column">
+    <div class="section-headline">
+      Build your AI chat app in a few lines of code...
     </div>
+    <iframe src="https://google.github.io/mesop/demo/?demo=embed/chat" class="main-demo-iframe box-shadow"></iframe>
   </div>
-</div>
+</section>
 
-<div class="mascot-image-spacer"></div>
-
-Mesop is a Python-based UI framework that allows you to rapidly build web apps like demos and internal apps:
-
-<div class="box-row">
-  <div class="reason-box">
-    <div class="reason-title">Easy to get started</div>
-    <ul>
-      <li>Write UI in <a href="#demo">idiomatic Python code</a></li>
-      <li>Skip the FE learning curve.</li>
-      <li>Ready to use components (e.g. <a href="./components/chat/">chat</a>)</li>
-    </ul>
-  </div>
-  <div class="reason-box">
-    <div class="reason-title">Fast iteration</div>
-    <ul>
-      <li><b>Hot reload</b> so the browser automatically reloads and preserves state</li>
-      <li>Rich IDE support with strong type safety</li>
-    </ul>
-  </div>
-  <div class="reason-box">
-    <div class="reason-title">Flexible & composable</div>
-    <ul>
-      <li>Build custom UIs <em>without</em> writing Javascript/CSS/HTML</li>
-      <li>Compose your UI into <a href="./components/">components</a>, which are just Python functions</li>
-    </ul>
+<section class="container">
+  <div class="section-chip teal">Python-native</div>
+  <div class="section-headline">
+    Write your UI in idiomatic Python
   </div>
-</div>
+  <div class="section-body">
+    With Mesop, you can leverage your existing Python expertise to build UIs effortlessly. No new languages to learn - just write clean, idiomatic Python using familiar constructs like functions, loops, and conditionals.
+  </div>
+</section>
 
-<h2 style="margin: 0.5rem"> See what you can build in less than 10 lines of code... </h2>
+=== "State management"
 
-<iframe class="immersive-demo" src="https://google.github.io/mesop/demo/"></iframe>
+    ``` python
+    @me.stateclass
+    class State:
+      val: str
 
-Check out how the above [demo gallery](./demo.md) was [built in pure Mesop](https://github.com/google/mesop/blob/main/demo/main.py)!
+    @me.page()
+    def page():
+        state = me.state(State)
+        me.text(state.val)
+    ```
 
-## Try it
+=== "Control flow"
 
-### Colab
+    ``` python
+    @me.stateclass
+    class State:
+      is_loaded: bool
 
-Try Mesop on Colab: [![Open In Colab](assets/colab.svg)](https://colab.research.google.com/github/google/mesop/blob/main/notebooks/mesop_colab_getting_started.ipynb)
+    @me.page()
+    def page():
+        if me.state(State).is_loaded:
+          me.text("Loaded")
+        else:
+          me.progress_spinner()
+    ```
 
-### Locally
+=== "Streaming UI"
 
-__Step 1:__ Install it
+    ```python
+    --8<-- "mesop/examples/docs/streaming.py"
+    ```
+
+=== "Composable"
+
+    ``` python
```

Comment:
I like showing how Mesop is Composable, but this example is not the clearest to me out of context. Maybe it's the use of "scaffolding"?